module-telemetry/src/main/java/com/mapbox/maps/module/telemetry/MapTelemetryImpl.kt

```diff
@@ -28,39 +34,78 @@ class MapTelemetryImpl : MapTelemetry {
   constructor(appContext: Context, accessToken: String) {
     this.appContext = appContext
     this.accessToken = accessToken
-    this.telemetry = MapboxTelemetry(appContext, accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT)
-    val telemetryState: TelemetryEnabler.State =
-      TelemetryEnabler.retrieveTelemetryStateFromPreferences()
-    if (TelemetryEnabler.State.ENABLED == telemetryState) {
-      telemetry.enable()
+
+    // check in the resources
+    val eventsTokenResId = getEventsResId(appContext, EVENTS_ACCESS_TOKEN_RESOURCE_NAME)
+    val eventsToken =
+      if (eventsTokenResId != 0) appContext.getString(eventsTokenResId) else accessToken
+
+    // TODO deferredDeliveryServiceOptions?
+    val eventsServiceOptions = EventsServerOptions(eventsToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, null)
```

Comment:
@tatiana-yan could you confirm that we can simply set null to the deferredDeliveryServiceOptions? Or do we need some extra logic here?

Reply:
I think `null` ok for now because deferred telemetry is a new feature and it is safe to just switch from MME to CoreTelemetry now and probably introduce deferred options later.
It's very new and I think we do not have clear guidelines for it now (it's in progress).
@DorkMatter WDYT?

Reply:
I talked to @DorkMatter, she confirmed that we need `null` here

Reply:
Thanks for the confirmation!